olp-cpp-sdk-core/src/cache/DiskCache.cpp

```diff
@@ -274,13 +270,30 @@ boost::optional<std::string> DiskCache::Get(const std::string& key) {
              : boost::none;
 }
 
-bool DiskCache::Remove(const std::string& key) {
+bool DiskCache::Remove(const std::string& key, uint64_t& removed_data_size) {
   if (!database_) {
+    removed_data_size = 0;
+    
     OLP_SDK_LOG_ERROR(kLogTag, "Remove: Database is uninitialized");
     return false;
   }
-  return database_->Delete(leveldb::WriteOptions(), key).ok();
-}
+
+  uint64_t data_size = 0u;
+  auto it = NewIterator({});
+  for (it->Seek(key); it->Valid(); it->Next()) {
+    if (it->key() == key) {
+      data_size = key.size() + it->value().size();
+      break;
+    }
+  }
```

Comment:
Here might be a bug, if the key does not exist, we will iterate over the rest of the keys.